src/Magnum/Math/Geometry/Intersection.h

```diff
@@ -166,28 +394,206 @@ template<class T> bool pointFrustum(const Vector3<T>& point, const Frustum<T>& f
     return true;
 }
 
-template<class T> bool boxFrustum(const Range3D<T>& box, const Frustum<T>& frustum) {
-    for(const Vector4<T>& plane: frustum.planes()) {
-        bool cornerHit = 0;
+template<class T> bool rangeFrustum(const Range3D<T>& box, const Frustum<T>& frustum) {
+    /* Convert to center/extent, avoiding division by 2 and instead comparing
+       to 2*-plane.w() later */
+    const Vector3<T> center = box.min() + box.max();
+    const Vector3<T> extent = box.max() - box.min();
 
-        for(UnsignedByte c = 0; c != 8; ++c) {
-            const Vector3<T> corner = Math::lerp(box.min(), box.max(), Math::BoolVector<3>{c});
+    for(const Vector4<T>& plane: frustum.planes()) {
+        const Vector3<T> absPlaneNormal = Math::abs(plane.xyz());
 
-            if(Distance::pointPlaneScaled<T>(corner, plane) >= T(0)) {
-                cornerHit = true;
-                break;
-            }
+        const Float d = Math::dot(center, plane.xyz());
+        const Float r = Math::dot(extent, absPlaneNormal);
+        if(d + r < -T(2)*plane.w()) {
+            return false;
         }
+    }
 
-        /* All corners are outside this plane */
-        if(!cornerHit) return false;
+    return true;
+}
+
+template<class T> bool aabbFrustum(const Vector3<T>& center, const Vector3<T>& extents, const Frustum<T>& frustum) {
+    for(const Vector4<T>& plane: frustum.planes()) {
+        const Vector3<T> absPlaneNormal = Math::abs(plane.xyz());
+
+        const Float d = Math::dot(center, plane.xyz());
+        const Float r = Math::dot(extents, absPlaneNormal);
+        if(d + r < -plane.w()) {
+            return false;
+        }
     }
 
-    /** @todo potentially check corners here to avoid false positives */
+    return true;
+}
+
+template<class T> bool sphereFrustum(
+    const Vector3<T>& center, const T radius, const Frustum<T>& frustum)
+{
```

Comment:
The declarations aren't too long to be wrapped like this, I think. Similarly in most cases below.

Reply:
From the Corrade [coding style](http://doc.magnum.graphics/corrade/corrade-coding-style.html) I read:

> vertical and horizontal compression, fitting more code on a screen while still making it possible to have more files open side by side by wrapping on the 80th column

I am pretty sure I only wrapped those which exceeded that quite a bit. You may need to specify "aren't too long" a bit more.

Reply:
This particular line is 105 colums long if not wrapped

Reply:
I did, however, miss quite a few (like the function above) it seems :D

Reply:
Okay, yeah, the coding style is a bit flaky here. In particular, I don't *demand* strict 78 column wrapping like for example the Python PEP-8 tools, I require that only for documentation blocks and bigger blocks of wide code, where the editor wrapping them would make the block significantly harder to read.

Function signatures being a bit longer are okay, because that's just a single line among usually short lines, unless they are so long that an editor wraps them on three and more lines, making editing and reviewing the changes harder.

All function parameter lists that are here wrapped on one or one and half line are still in the okay range to have the full signature in a single line, I think.

Reply:
As in

~~~
someFunc(
    This is, One line)
{
~~~

and

~~~
SomeOtherFunc(
    That is, Clearly way, Too long,
    Not to, Be broken) //two lines? One and a half?
{
~~~